url8f84d8d956_klapaukh_JTuttePoly_tgz-pJ8-BitTutteJ8/src/selection/HeuristicCollection.java

```diff
@@ -0,0 +1,150 @@
+package selection;
+
+import graph.Graph;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import util.Debug;
+import util.Pair;
+import util.Triple;
+
+public class HeuristicCollection implements EdgeSelection{
+
+	public int edge_selection_heuristic;
+
+	public static final int RANDOM = 0;
+	public static final int CUT = 1;
+	public static final int VERTEX_ORDER = 2;
+	public static final int MAXIMISE_SDEGREE = 3;
+	public static final int MAXIMISE_MDEGREE = 4;
+	public static final int MINIMISE_DEGREE = 5;
+	public static final int MINIMISE_SDEGREE = 6;
+	public static final int MINIMISE_MDEGREE = 7;
+	public static final int MAXIMISE_DEGREE = 8;
+	
+	public HeuristicCollection(int heuristic){
+		this.edge_selection_heuristic = heuristic;
+	}
+
+	public Triple<Integer, Integer, Integer> select_edge(Graph graph, boolean reduce_multiedges) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.